pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/LocalPinotFS.java

```diff
@@ -64,18 +64,26 @@ public boolean delete(URI segmentUri) throws IOException {
   }
 
   @Override
-  public boolean move(URI srcUri, URI dstUri) throws IOException {
+  public boolean move(URI srcUri, URI dstUri, boolean overwrite) throws IOException {
     File srcFile = new File(decodeURI(srcUri.getRawPath()));
     File dstFile = new File(decodeURI(dstUri.getRawPath()));
     if (dstFile.exists()) {
-      FileUtils.deleteQuietly(dstFile);
+      if (overwrite) {
+        FileUtils.deleteQuietly(dstFile);
+      } else {
+        // dst file exists, returning
+        return false;
+      }
     }
+
+    // Makes all parent directories for dst file if they don't exist
     if (!srcFile.isDirectory()) {
```

Comment:
Not related to this one, but why do we need this check?

Reply:
we want to make all relevant parent directories for dstURI if they don't exist. Also, FileUtils has different methods to move files v. directories.

Reply:
My question is that, why not always use **Files.move()**?